### PR TITLE
Use vertex buffer objects

### PR DESCRIPTION
**Pull request by ascent12**
These are just some pretty mundane changes to the GLES renderer. In my extremely limited testing, using VAOs/VBOs instead of uploading the same vertex data over and over lowers the CPU usage _slightly_. There is still a lot more to do, but this is a start.

There are also some minor changes to cleanup the shaders.

### CODE REVIEW

**Comment by SirCmpwn** [comment]
I'm not sure these extensions are going to be consistently available. VAOs are a GLES3 feature. I have a laptop that doesn't support GLES3 I'll have to test this with.

**Comment by SirCmpwn** [comment]
That was the main motivator behind switching from gles3 to gles2 by the way.

File: render/gles2/shaders.c [hunk 1]
```diff
@@ -3,100 +3,73 @@
...
-"    vec4 i1 = inMatrix[1];"
-"    vec4 i2 = inMatrix[2];"
-"    vec4 i3 = inMatrix[3];"
-"    mat4 outMatrix = mat4("
-"                 vec4(i0.x, i1.x, i2.x, i3.x),"
-"                 vec4(i0.y, i1.y, i2.y, i3.y),"
-"                 vec4(i0.z, i1.z, i2.z, i3.z),"
-"                 vec4(i0.w, i1.w, i2.w, i3.w)"
-"                 );"
-"    return outMatrix;"
```
Comment (SirCmpwn) [inline render/gles2/shaders.c]: Where did this logic go?

**Comment by ascent12** [comment]
Mesa supports these on every driver (see https://github.com/mesa3d/mesa/blob/master/docs/features.txt), considering they're a required feature of desktop OpenGL 3.0.
A lot of the GL extensions are ubiquitous,  so I think they're fine to use.

File: render/gles2/shaders.c (hunk 1 above)
Comment (ascent12) [reply render/gles2/shaders.c]: https://github.com/swaywm/wlroots/pull/290/commits/7170170a5778a0bbb37827f85882cde4c7762ec5#diff-650ac2c117ab1d2399640eab8ba3ab74R170

Comment (SirCmpwn) [reply render/gles2/shaders.c]: Hah, derp. Good catch.

**Comment by SirCmpwn** [comment]
I was right - this doesn't work on older hardware. I tested this on a ThinkPad X200 to no avail.

I'm 100% on board with VAOs, but we need to fall back. I think it would be better to implement a gles3 renderer and choose whichever one is most suitable for the hardware (gles3 will fail to initialize on older machines, so we can just ditch it and switch to gles2 and it should Just Werk).

**Comment by SirCmpwn** [comment]
We used to have a gles3 backend, by the way, should be fairly straightforward to bring it back to life.

**Comment by ascent12** [comment]
I'm pretty sure most of the real advantages are coming from using VBOs, and the VAOs only making it nicer to use, so I removed them.
It really is annoying having to limit it to GLES2, but I really can't be bothered writing multiple ways of trying to do the same thing.

**Comment by SirCmpwn** [comment]
That doesn't work either. You have to actually test if it's supported before you use it, or make a separate gles3 renderer.

**Comment by ascent12** [comment]
VBOs are standard GLES2.
I'm not 100% I'm using them "properly", since I've never used them without VAOs before, but the code did work for me.

**Comment by SirCmpwn** [comment]
I will spend some more time with it and see what I can find out.

**Comment by ascent12** [comment]
The non-VBO changes have been integrated into #319

**Pull request closed without merging**